common/app/services/newsletters/NewsletterApi.scala

```diff
@@ -0,0 +1,95 @@
+package services.newsletters
+
+import com.gu.identity.model.{EmailEmbed, NewsletterIllustration}
+import common.{BadConfigurationException, Logging}
+import conf.Configuration._
+import play.api.libs.json.{JsResult, JsValue, Json}
+import play.api.libs.ws.WSClient
+
+import scala.concurrent.duration.DurationInt
+import scala.concurrent.{ExecutionContext, Future}
+
+case class NewsletterResponse(
+    id: String,
+    name: String,
+    brazeNewsletterName: String,
+    brazeSubscribeAttributeName: String,
+    brazeSubscribeEventNamePrefix: String,
+    theme: String,
+    description: String,
+    frequency: String,
+    exactTargetListId: Int,
+    emailEmbed: EmailEmbed,
+    illustration: Option[NewsletterIllustration] = None,
+)
+
+case class GroupedNewsletterResponse(
+    displayName: String,
+    newsletters: List[NewsletterResponse],
+)
+
+case class GroupedNewslettersResponse(
+    newsRoundups: GroupedNewsletterResponse,
+    newsByTopic: GroupedNewsletterResponse,
+    features: GroupedNewsletterResponse,
+    sport: GroupedNewsletterResponse,
+    culture: GroupedNewsletterResponse,
+    lifestyle: GroupedNewsletterResponse,
+    comment: GroupedNewsletterResponse,
+    work: GroupedNewsletterResponse,
+    fromThePapers: GroupedNewsletterResponse,
+)
+
+case class NewsletterApi(wsClient: WSClient)(implicit executionContext: ExecutionContext)
+    extends Logging
+    with implicits.WSRequests {
+
+  implicit val emailEmbedReads = Json.reads[EmailEmbed]
+
+  implicit val newsletterIllustrationReads = Json.reads[NewsletterIllustration]
+
+  implicit val newsletterResponseReads = Json.reads[NewsletterResponse]
+
+  implicit val groupedNewsletterResponseReads = Json.reads[GroupedNewsletterResponse]
+
+  implicit val groupedNewslettersReads = Json.reads[GroupedNewslettersResponse]
+
+  private def ensureHostSecure(host: String): String = host.replace("http:", "https:")
+
+  private def getBody(path: String): Future[JsValue] = {
+    val maybeJson: Option[Future[JsValue]] = for {
+      host <- newsletterApi.host
+      origin <- newsletterApi.origin
+    } yield {
+      val url = s"${ensureHostSecure(host)}/$path"
+      log.info(s"Making request to newsletters API: $url")
+      wsClient
+        .url(url)
+        .withRequestTimeout(10.seconds)
+        .withHttpHeaders(("Origin", origin))
+        .getOKResponse()
+        .map(_.json)
+    }
+
+    maybeJson.getOrElse(
+      Future.failed(new BadConfigurationException("Newsletters API host or origin not configured")),
+    )
+  }
+
+  def getGroupedNewsletters(): Future[JsResult[GroupedNewslettersResponse]] = {
```

Comment:
I don't think `getGroupedNewsletters` and `getNewsletters` should return a JsResult. For the user of that service, it shouldn't matter that the underlying API is producing Json, what does matter are the case classes you've defined above. I think the academic term for this is called "Leaky abstraction"

You might want to consider returning `Future[Either[String, GroupedNewslettersResponse]]` or `Future[Either[NewsletterError, GroupedNewslettersResponse]]` where you define `NewsletterError` as being a new type.

Then suddenly, the code in your agent doesn't have to handle JSON parsing errors, and it should bubble up all the way to you controller where I've seen you've added some logging for that case. You'll still have logging there, but it will be for a `NewsletterApiError` or such